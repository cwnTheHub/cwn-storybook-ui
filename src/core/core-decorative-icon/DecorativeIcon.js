import React from "react";
import PropTypes from "prop-types";
import Icon from "../../utils/components/Icon/Icon";

const DecorativeIcon = ({ symbol, variant, size, ...rest }) => {
  return (
    <Icon
      {...rest}
      symbol={symbol}
      variant={variant}
      size={size}
      aria-hidden="true"
    />
  );
};

DecorativeIcon.propTypes = {
  symbol: PropTypes.oneOf([
    "caretDown",
    "caretUp",
    "checkmark",
    "chevron",
    "leftChevron",
    "exclamationPointCircle",
    "expander",
    "hamburger",
    "location",
    "minus",
    "plus",
    "questionMarkCircle",
    "spyglass",
    "times",
  ]).isRequired,
  variant: PropTypes.oneOf(["primary", "secondary", "inverted", "error"]),
  size: PropTypes.oneOf([16, 20, 24, 32, 48]),
};

DecorativeIcon.defaultProps = {
  variant: undefined,
  size: 24,
};

export default DecorativeIcon;
