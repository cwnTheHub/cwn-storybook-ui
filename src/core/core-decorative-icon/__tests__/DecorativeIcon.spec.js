import React from "react";
import { shallow, render } from "enzyme";

import DecorativeIcon from "../DecorativeIcon";

describe("DecorativeIcon", () => {
  const defaultProps = {
    symbol: "spyglass",
  };
  const doShallow = (props = {}) =>
    shallow(<DecorativeIcon {...defaultProps} {...props} />);

  it("renders", () => {
    const icon = render(<DecorativeIcon symbol="spyglass" />);

    expect(icon).toMatchSnapshot();
  });

  it("passes attributes to the Icon component", () => {
    const icon = doShallow({
      symbol: "checkmark",
      variant: "secondary",
      size: 16,
      id: "the-icon",
    });

    expect(icon.props()["symbol"]).toBe("checkmark");
    expect(icon.props()["variant"]).toBe("secondary");
    expect(icon.props().size).toBe(16);
    expect(icon.props().id).toBe("the-icon");
  });

  it("is hidden from screen readers", () => {
    const icon = doShallow();

    expect(icon.props()["aria-hidden"]).toBe("true");
  });
});
