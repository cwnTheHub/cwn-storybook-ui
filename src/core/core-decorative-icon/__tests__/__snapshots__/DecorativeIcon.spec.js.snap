// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecorativeIcon renders 1`] = `
LoadedCheerio {
  "0": <i
    aria-hidden="true"
    class="sc-beqWaB jaKVZE"
  />,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
